base/strings/substring.jl

```diff
@@ -225,13 +225,9 @@ end
 end
 
 # nothrow needed here because for v in a can't prove the indexing is inbounds.
-@assume_effects :foldable :nothrow function string(a::Union{Char, String, Symbol}...)
-    _string(a...)
-end
+@assume_effects :foldable :nothrow string(a::Union{Char, String, Symbol}...) = _string(a...)
 
-function string(a::Union{Char, String, SubString{String}, Symbol}...)
-    _string(a...)
-end
+string(a::SubString{String}...) = _string(a...)
```

Comment:
These are not equivalent, since the former specified a list of any combination of them, while this specifies a list of only substrings.

Reply:
Right, thanks for pointing it out.